Maple2.Server.Game/Scripting/Npc/NpcScriptContext.cs

```diff
@@ -98,6 +85,37 @@ public class NpcScriptContext {
         return true;
     }
 
+    public void SetTalkTypeFlags(int firstState) {
```

Comment:
I feel like this function can be reorganized to better accommodate quests in the future:

```cs
switch (Npc.Value.Metadata.Basic.Kind) {
    int options = 0;
    case 1 or > 10 and < 20:
        TalkType |= NpcTalkType.Dialog;
        options++;
        break;
    case ...
        TalkType |= NpcTalkType.Dialog
}

if (Metadata.States.TryGetValue(firstState, out ScriptState? scriptState)) {
    if (scriptState.Type == ScriptStateType.Job) {
        TalkType |= NpcTalkType.Dialog;
    } else {
        TalkType |= NpcTalkType.Talk;
        options++;
    }
}

if (options > 1) {
    TalkType |= NpcTalkType.Select;
}
````

Reply:
Not sure if the logic is 100% accurate. Is a Job script counted separate from first script? (e.g. can an Npc have both and does that require Select?)